util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,264 @@
+    private OptionalDouble tryDoubleParse(String line, Predicate<? super Double> doublePredicate) {
+        try {
+            var num = Double.parseDouble(line);
+            return doublePredicate.test(num) ? OptionalDouble.of(num) : OptionalDouble.empty();
+        }
+        catch (NumberFormatException e) {
+            return OptionalDouble.empty();
+        }
+    }
+
+    /**
+     * Gets the next yes or no answer.
+     * Calls {@link InputUtility#nextString(String, Predicate)}.
+     * Passes the prompt and a predicate that validates "y", "n", "yes", "no".
+     *
+     * @param prompt the prompt to display to the user.
+     * @return true for yes and false for no.
+     * @throws IOException if there's a problem while reading from the underlying stream.
+     */
+    public boolean nextYesNo(String prompt) throws IOException {
+        var result = this.nextString(prompt, yesOrNo()).trim().toLowerCase();
+        return YES_SYNONYMS.contains(result);
+    }
+
+    /**
+     * Waits for the user to press &lt;enter&gt;.
+     * The prompt defaults to "Press when ready".
+     * Calls {@link InputUtility#pause(String)}.
+     *
+     * @throws IOException if there's a problem while reading from the underlying stream.
+     */
+    public void pause() throws IOException {
+        pause("Press enter when ready");
+    }
+
+    /**
+     * Uses prompt entered and waits for the user to press &lt;enter&gt;.
+     * Calls nextString{@link InputUtility#nextString(String, Predicate)}.
+     *
+     * @param prompt the prompt to display to the user.
+     * @throws IOException if there's a problem while reading from the underlying stream.
+     */
+    public void pause(String prompt) throws IOException {
+        this.nextString(prompt, String::isBlank);
+    }
+
+    /**
+     * Get a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     *
+     * @return a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     */
+    public static Predicate<String> yesOrNo() {
+        return s -> {
+            var cleanedString = s.trim().toLowerCase();
+            return YES_SYNONYMS.contains(cleanedString) || NO_SYNONYMS.contains(cleanedString);
+        };
+    }
+
+
+    /**
+     * Get a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     *
+     * @param firstOption the first of the strings to match to.
+     * @param otherOptions the strings for the input to match to.
+     * @return a String predicate that takes one or more Strings and tests whether any one of them match the input.
+     */
+    public static Predicate<String> oneOfThese(String firstOption, String... otherOptions) {
+        return s -> Stream.concat(Arrays.stream(otherOptions), Stream.of(firstOption))
+            .anyMatch(choice -> choice.equalsIgnoreCase(s.trim()));
+    }
+
+    /**
+     * Get an int predicate that tests whether the input is between the two limits.
```

Comment:
Start method descriptions with a verb in present tense, not imperative mood:

```suggestion
     * Gets an int predicate that tests whether the input is between the two limits.
```

Fix this in other methods as well.